services/datamanager/builtin/builtin.go

```diff
@@ -450,10 +450,11 @@ func (svc *builtIn) Update(ctx context.Context, cfg *config.Config) error {
 		modelsToDeploy := svcConfig.ModelsToDeploy
 		errorChannel := make(chan error, len(modelsToDeploy))
 		go svc.modelManager.DownloadModels(cfg, modelsToDeploy, errorChannel)
+		// close(errorChannel)
 		if len(errorChannel) != 0 {
 			var s string
-			for i := range errorChannel {
-				s = s + "\n" + i.Error()
+			for err := range errorChannel {
+				s = s + "," + err.Error()
```

Comment:
This would cause a comma to prepend the initial error. Adding it after would add one-too-many commas at the end.
So instead you could do:

```
var errMsgs []string
for err := range errorChannel {
   errMsgs = append(errMsgs, err.Error()
}
return Errors.New(strings.Join(errMsgs[:], ", "))
```
If it's empty, it returns an empty string. And then joins with ", " for len(errMsgs) > 1

Reply:
changed.